trendly/src/app/histogram-section/histogram-section.component.ts

```diff
@@ -0,0 +1,161 @@
+import {Component, HostListener, Input, OnInit, SimpleChanges} from '@angular/core';
+
+import {ColorsService} from '../colors.service';
+
+interface chartRole {
+  role: string;
+}
+type ColumnArray = (chartRole|string)[];
+
+interface Topic {
+  name: string;
+  volume: number;
+  description: string;
+}
+
+interface DataType {
+  [index: string]: Array<Topic>;
```

Comment:
Use [] for simple array types
```suggestion
  [index: string]: Topic[];
```

Reply:
Done.